src/Symfony/Component/HttpClient/ApiClient.php

```diff
@@ -0,0 +1,136 @@
+     * @param array $defaultOptions     Default requests' options
+     * @param int   $maxHostConnections The maximum number of connections to a single host
+     *
+     * @see HttpClientInterface::OPTIONS_DEFAULTS for available options
+     */
+    public static function create(array $defaultOptions = [], int $maxHostConnections = 6): ApiClientInterface
+    {
+        return new self(HttpClient::create($defaultOptions, $maxHostConnections));
+    }
+
+    public function __construct(HttpClientInterface $client, array $defaultOptions = [])
+    {
+        $this->client = $client;
+
+        if ($defaultOptions) {
+            [, $this->defaultOptions] = self::prepareRequest(null, null, $defaultOptions, $this->defaultOptions);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function get(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('GET', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function head(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('HEAD', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function post(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('POST', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function put(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('PUT', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function patch(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('PATCH', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function delete(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('DELETE', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function options(string $url, iterable $options = []): ApiResponseInterface
+    {
+        return $this->requestApi('OPTIONS', $url, $options);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function complete($responses, float $timeout = null): ApiResponseStreamInterface
+    {
+        if ($responses instanceof ApiResponse) {
+            $responses = [$responses];
+        } elseif (!\is_iterable($responses)) {
+            throw new \TypeError(sprintf('%s() expects parameter 1 to be iterable of ApiResponse objects, %s given.', __METHOD__, \is_object($responses) ? \get_class($responses) : \gettype($responses)));
+        }
+
+        return new ApiResponseStream(ApiResponse::complete($this->client, $responses, $timeout));
+    }
+
+    private function requestApi(string $method, string $url, iterable $options): ApiResponse
+    {
+        $options['buffer'] = true;
+        [, $options] = self::prepareRequest(null, null, $options, $this->defaultOptions);
```

Comment:
this method requires array, iterable passed

Reply:
Thanks. All `iterable` type hints where leftovers, now fixed.